From b20c97f58f9eca426a69724748c4d75919668f29 Mon Sep 17 00:00:00 2001
From: ChenXiaoSong <chenxiaosong@chenxiaosong.com>
Date: Fri, 31 May 2024 01:29:39 +0000
Subject: [PATCH] debug xfs log

Signed-off-by: ChenXiaoSong <chenxiaosong@chenxiaosong.com>
---
 fs/xfs/xfs_log.c | 88 ++++++++++++++++++++++++++++++++++++++++++++++--
 1 file changed, 86 insertions(+), 2 deletions(-)

diff --git a/fs/xfs/xfs_log.c b/fs/xfs/xfs_log.c
index 416c15494983..075eb331686c 100644
--- a/fs/xfs/xfs_log.c
+++ b/fs/xfs/xfs_log.c
@@ -21,6 +21,9 @@
 #include "xfs_sb.h"
 #include "xfs_health.h"
 
+int debug_xfs_log = 0;
+EXPORT_SYMBOL_GPL(debug_xfs_log);
+
 struct kmem_cache	*xfs_log_ticket_cache;
 
 /* Local miscellaneous function prototypes */
@@ -1862,6 +1865,83 @@ xlog_map_iclog_data(
 	return 0;
 }
 
+/**
+ * xfs_memscan - Find a character in an area of memory.
+ * @addr: The memory area
+ * @c: The byte to search for
+ * @size: The size of the area.
+ *
+ * returns the address of the first occurrence of @c,
+ * or NULL if @c is not found
+ */
+static void *xfs_memscan(void *addr, int c, size_t size)
+{
+	unsigned char *p = addr;
+
+	while (size) {
+		if (*p == (unsigned char)c)
+			return (void *)p;
+		p++;
+		size--;
+	}
+	return NULL;
+}
+
+/*
+ * return 0 if bio have special data,
+ * or 1 if it is not found
+ */
+static int check_bio_data(struct bio *bio) {
+	int ret = 1;
+	struct bio_vec *bvec;
+	struct bvec_iter_all iter_all;
+	char *kaddr;
+	char *tmp;
+	char *last_addr;
+	unsigned int len;
+
+	bio_for_each_segment_all(bvec, bio, iter_all) {
+		kaddr = kmap_local_page(bvec->bv_page);
+		last_addr = kaddr + bvec->bv_offset + bvec->bv_len - 1;
+		tmp = kaddr + bvec->bv_offset;
+		goto scan;
+again:
+		tmp++;
+scan:
+		len = last_addr - tmp + 1;
+		tmp = xfs_memscan(tmp, 'X', len);
+		if (!tmp)
+			goto unmap;
+		// printk("%s:%d, find X\n", __func__, __LINE__);
+		if (!(tmp + 1 <= last_addr && tmp[1] == 'A'))
+			goto again;
+		// printk("%s:%d, find A\n", __func__, __LINE__);
+		if (!(tmp + 2 <= last_addr && tmp[2] == 'G'))
+			goto again;
+		// printk("%s:%d, find G\n", __func__, __LINE__);
+		if (!(tmp + 3 <= last_addr && tmp[3] == 'F'))
+			goto again;
+		// printk("%s:%d, find F\n", __func__, __LINE__);
+		// printk("%s:%d, find success\n", __func__, __LINE__);
+		ret = 0;
+unmap:
+		kunmap_local(kaddr);
+		if (!ret)
+			return ret;
+	}
+	return ret;
+}
+
+static void xfs_submit_bio(struct bio *bio)
+{
+	if (debug_xfs_log && !check_bio_data(bio)) {
+		printk("%s:%d, exec xlog_bio_end_io instead of submit_bio\n", __func__, __LINE__);
+		xlog_bio_end_io(bio);
+	} else {
+		submit_bio(bio);
+	}
+}
+
 STATIC void
 xlog_write_iclog(
 	struct xlog		*log,
@@ -1943,13 +2023,13 @@ xlog_write_iclog(
 		split = bio_split(&iclog->ic_bio, log->l_logBBsize - bno,
 				  GFP_NOIO, &fs_bio_set);
 		bio_chain(split, &iclog->ic_bio);
-		submit_bio(split);
+		xfs_submit_bio(split);
 
 		/* restart at logical offset zero for the remainder */
 		iclog->ic_bio.bi_iter.bi_sector = log->l_logBBstart;
 	}
 
-	submit_bio(&iclog->ic_bio);
+	xfs_submit_bio(&iclog->ic_bio);
 	return;
 shutdown:
 	xlog_force_shutdown(log, SHUTDOWN_LOG_IO_ERROR);
@@ -2253,6 +2333,10 @@ xlog_write_full(
 		struct xfs_log_iovec	*reg = &lv->lv_iovecp[index];
 		struct xlog_op_header	*ophdr = reg->i_addr;
 
+		char *log_data = (char *)ophdr + sizeof(struct xlog_op_header);
+		printk("%s:%d, 0x%02x 0x%02x 0x%02x 0x%02x\n", __func__, __LINE__,
+		       log_data[0], log_data[1], log_data[2], log_data[3]);
+
 		ophdr->oh_tid = cpu_to_be32(ticket->t_tid);
 		xlog_write_iovec(iclog, log_offset, reg->i_addr,
 				reg->i_len, len, record_cnt, data_cnt);
-- 
2.34.1

