From 9015acfdcd861b80fe8d321d5e03c2cfcdd9ac4f Mon Sep 17 00:00:00 2001
From: ChenXiaoSong <chenxiaosong@kylinos.cn>
Date: Mon, 15 Dec 2025 16:06:09 +0800
Subject: [PATCH] smb: include mydebug.h

Signed-off-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
---
 fs/smb/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs/smb/Makefile b/fs/smb/Makefile
index 353b1c2eefc4..58c430b901c7 100644
--- a/fs/smb/Makefile
+++ b/fs/smb/Makefile
@@ -1,5 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 
+subdir-ccflags-y += -include $(srctree)/include/mydebug.h
+
 obj-$(CONFIG_SMBFS)		+= common/
 obj-$(CONFIG_SMBDIRECT)		+= smbdirect/
 obj-$(CONFIG_CIFS)		+= client/
-- 
2.54.0

